### ISSUE TITLE
libvoreutils.hpp: build warning on armel/armhf/arm64

### ISSUE THREAD

**Issue by a-detiste** [systemd-cron/systemd-cron#105]
```
/<<PKGBUILDDIR>>/src/bin/crontab.cpp: In function ‘int main(int, char* const*)’:
/<<PKGBUILDDIR>>/src/bin/crontab.cpp:405:86: **warning: comparison is always true due to limited range of data type [-Wtype-limits]**
  405 |         for(char arg; (arg = getopt_long(argc, argv, "lriestu:", longopts, nullptr)) != -1;)
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
```

https://buildd.debian.org/status/fetch.php?pkg=systemd-cron&arch=armhf&ver=1.90-1&stamp=1691962896&raw=0

```
In file included from /usr/include/c++/13/map:62,
                 from /<<PKGBUILDDIR>>/out/build/include/libvoreutils.hpp:20:
/usr/include/c++/13/bits/stl_tree.h: In member function ‘std::pair<std::_Rb_tree_node_base*, std::_Rb_tree_node_base*> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_get_insert_hint_unique_pos(const_iterator, const key_type&) [with _Key = std::__cxx11::basic_string<char>; _Val = std::pair<const std::__cxx11::basic_string<char>, long long unsigned int>; _KeyOfValue = std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, long long unsigned int> >; _Compare = std::less<std::__cxx11::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::__cxx11::basic_string<char>, long long unsigned int> >]’:

/usr/include/c++/13/bits/stl_tree.h:2210:5: note: parameter passing for argument of type ‘std::_Rb_tree<std::__cxx11::basic_string<char>, std::pair<const std::__cxx11::basic_string<char>, long long unsigned int>, std::_Select1st<std::pair<const std::__cxx11::basic_string<char>, long long unsigned int> >, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, long long unsigned int> > >::const_iterator’ changed in GCC 7.1
 2210 |     _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by nabijaczleweli** [comment]
The first one: I think these have char=unsigned char; this should just be int!! sorry for not testing this before on a char=uchar platform, i don't have an arm64 platform rn, they should have aarch64 builders on salsa IMO. Trivial fix, looks like you already did it in 17ac11899cbbd878a7df659f2bcc24c16f8b5c6b

Second one: voreutils is a red herring since it includes `<map>` transitively in the PCH. Looks like it's just a warning: https://stackoverflow.com/questions/48149323/what-does-the-gcc-warning-project-parameter-passing-for-x-changed-in-gcc-7-1-m because we're using `std::map<string, u64>::operator[]` and it returns an `u64&`(?) in case we were embedding this in our ABI as a library, and had compatibility against gcc <5: we aren't, so we don't care. f2ff31bf47813ad2ecb98f231169748d9065ffbd

Tested on [builds.sr.ht debian/sid arch=arm64 CI](https://builds.sr.ht/~nabijaczleweli/job/1041058). Can't repro the second issue there, but both clang and gcc accept -Wno-psabi and the gcc release says that's what's wrong, so closing in hopes it actually does work.

**Comment by a-detiste** [comment]
>  they should have aarch64 builders on salsa IMO

It was there, but that's so much stuff to look around
https://salsa.debian.org/nabijaczleweli/sd-cron-tmp/-/jobs/4551095#L1273
